### ISSUE TITLE
Break Python backward compatibility for DEAP 2.0 - Python 3.7+ all the way

### ISSUE THREAD

**Issue by cmd-ntrf** [DEAP/deap#290]
Mandatory thought provocating title. 

This up for debate, but I think that DEAP 2.0 might be ready for a serious change of pace and moving to supporting only Python 3.7+ could be the right kick in the butt we need. I think 2.0 is not about new features, but mainly refactoring based on the numerous user feedback we got since DEAP 1.0 release.

What do you think?

**Comment by fmder** [comment]
I totally agree that we shouldn't concern ourselfes with backward compatibility for old Python versions.

Do you have any feature you intend to use in 3.7 that is not in 3.6?

**Comment by cmd-ntrf** [comment]
Python 3.7 has data classes that could either simplify the creator or maybe even remove it completely.
https://www.python.org/dev/peps/pep-0557/

**Comment by TristanJM** [comment]
I am also in agreement with moving to Python 3. I would say 3.7 is a step too far considering you still require 3.6 to run Tensorflow.

**Comment by nielsgl** [comment]
TensorFlow already supports 3.7 for a couple of months now.

**Comment by fmder** [comment]
@cmd-ntrf I think I'm onto something here. No more creator, no more fitness.values, easy multi-crossover.

https://gist.github.com/fmder/a8eb232e8b51a56a300b8494d5255622

**Comment by Seanny123** [comment]
@fmder I really like that approach. It looks like would resolve #480

**Comment by Seanny123** [comment]
@fmder have you already started a branch working on this? Alternatively, what would a Minimal Viable PR look like for this?

**Comment by fmder** [comment]
@Seanny123 I really appreciate your enthusiasm, but I think this is too mush of a big change for a PR ;).

**Comment by Seanny123** [comment]
Months later, I still think about this issue :heart: 

Given this is marked as "in progress" is there anything I can do to help?

**Comment by nbro** [comment]
@cmd-ntrf, @fmder Is this issue going to be discussed for how many years before something is actually done?
